src/Microsoft.Extensions.Hosting/HostBuilder.cs

```diff
@@ -179,6 +179,7 @@ private void CreateServiceProvider()
             services.AddSingleton(_appConfiguration);
             services.AddSingleton<IApplicationLifetime, ApplicationLifetime>();
             services.AddSingleton<IHostLifetime, ProcessLifetime>();
+            services.AddSingleton<IHost, Host>();
```

Comment:
Don't these need to be after AddOptions?

Reply:
It's not order sensitive.